src/mods/visual/game.ts

```diff
@@ -37,3 +38,21 @@ extend(Room, {
 		},
 	},
 });
+
+declare module 'xxscreeps/game/map' {
+	interface GameMap {
+		/**
+		 * A `RoomVisual` object for the map. You can use this object to draw simple shapes (lines,
+		 * circles, text labels).
+		 */
+		readonly visual: RoomVisual;
+	}
+}
+
+extend(GameMap, {
+	visual: {
+		get() {
+			return new RoomVisual('map');
```

Comment:
Is there a reason not to memoize the visual instance like `Room` does (above, line 34)

Reply:
I actually had it that way but I think it won't work because the map is not recreated every tick? I was getting the same RoomVisual Object and it grew with every tick.

Reply:
I did not really found a solution with that... it works with getting a new instance on every call... To optimize it we would need to memoize the instance, but get a new one on first call every tick.
I looked through the code and thought about creating one with the `gameInitializer` hook but not sure if that makes sense... or maybe store it in a different object but I did not really find out where to store it then.